### PR TITLE
Changed artifact download URL pattern to match the new GraalVM CE release 19.0.0 TGZ file name standard

### PR DESCRIPTION
**Pull request by sanjeeva**
As of version 19.0.0 GraalVM team has changed the way the distro file is named in the GitHub download page for its Community Edition. Most notable is the change to the order of [version], [os] and [arch] as well as the use of "darwin" instead of "macos" for MacOS builds.

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/gradle-graal](https://github.com/palantir/gradle-graal), @sanjeeva! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

**Comment by uschi2000** [comment]
Do you know what the future versioning scheme will be? Looking https://www.graalvm.org/docs/release-notes/ I would presume they're mostly done with releasing -rc versions. If that's true then I would prefer removing support for rc versions in this plugin instead of adding the optionality.

**Comment by sanjeeva** [comment]
> Do you know what the future versioning scheme will be? Looking https://www.graalvm.org/docs/release-notes/ I would presume they're mostly done with releasing -rc versions. If that's true then I would prefer removing support for rc versions in this plugin instead of adding the optionality.

**Comment by sanjeeva** [comment]
A very valid comment @uschi2000, though the new release version has made a major change on how class initialization is done, they've now switched it from build time to run-time default init, with the option to switch it via flags.
Therefore it might be worth having some backward compatibility for a while.

**Comment by uschi2000** [comment]
I'm fine with that too. Happy to merge once the tests pass. Would you be able to add two tests clarifying the class init behavior described above?

File: src/main/java/com/palantir/gradle/graal/DownloadGraalTask.java [hunk 1]
```diff
@@ -51,15 +54,22 @@ public DownloadGraalTask() {
     public final void downloadGraal() throws IOException {
         Path cache = getCacheSubdirectory().get();
         Files.createDirectories(cache);
-        try (InputStream in = new URL(render(ARTIFACT_PATTERN)).openStream()) {
+        
+        final String artefactPattern = isGraalRcVersion() ?
```
Comment (yschimke) [inline src/main/java/com/palantir/gradle/graal/DownloadGraalTask.java:58]:
```suggestion
        final String artifactPattern = isGraalRcVersion() ? 
```

File: src/main/java/com/palantir/gradle/graal/DownloadGraalTask.java [hunk 2]
```diff
@@ -51,15 +54,22 @@ public DownloadGraalTask() {
     public final void downloadGraal() throws IOException {
         Path cache = getCacheSubdirectory().get();
         Files.createDirectories(cache);
-        try (InputStream in = new URL(render(ARTIFACT_PATTERN)).openStream()) {
+        
+        final String artefactPattern = isGraalRcVersion() ? 
+            ARTIFACT_PATTERN_RC_VERSION : ARTIFACT_PATTERN_RELEASE_VERSION;
+        
+        try (InputStream in = new URL(render(artefactPattern)).openStream()) {
```
Comment (yschimke) [inline src/main/java/com/palantir/gradle/graal/DownloadGraalTask.java:61]:
```suggestion
        try (InputStream in = new URL(render(artifactPattern)).openStream()) {
```

File: src/main/java/com/palantir/gradle/graal/DownloadGraalTask.java [hunk 3]
```diff
@@ -33,8 +33,11 @@
 /** Downloads GraalVM binaries. */
 public class DownloadGraalTask extends DefaultTask {
 
-    private static final String ARTIFACT_PATTERN = "[url]/vm-[version]/graalvm-ce-[version]-[os]-[arch].tar.gz";
-    private static final String FILENAME_PATTERN = "graalvm-ce-[version]-[arch].tar.gz";
```
Comment (carterkozak) [inline src/main/java/com/palantir/gradle/graal/DownloadGraalTask.java:37]: I'm not sure that we need to change the local filename. Supportive of adding the OS to match the URL, but I don't think this needs to differ based on the requested version.

**Comment by carterkozak** [comment]
@sanjeeva @yschimke @uschi2000 Thoughts on this? #118

**Comment by carterkozak** [comment]
Merged here: https://github.com/palantir/gradle-graal/pull/118

Looking into fixing the nativeImage task in a separate pr

**Pull request closed without merging**